src/app/cli/src/cli_value_parser.rs

```diff
@@ -113,3 +132,17 @@ pub(crate) fn value_parse_url(url_str: &str) -> Result<Url, String> {
         }
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub(crate) fn value_parse_dataset_visibility(value: &str) -> Result<DatasetVisibility, String> {
+    let res = match value {
+        "private" => DatasetVisibility::Private,
+        "public" => DatasetVisibility::PubliclyAvailable,
```

Comment:
Again, this unlikely is a responsibility of CLI app, could be enum's implementation detail or some implemented conversion trait

Reply:
In this case, I disagree: this is validation of values coming from user input, which refer specifically to CLI parameters